src/stores/payment.ts

```diff
@@ -38,9 +39,17 @@ export const usePaymentStore = defineStore('payment', {
   },
   actions: {
     priceInEuros() {
+      if (isNaN(this.tip)) {
+        this.tip = 0
+      }
+      if (isNaN(this.sum)) {
+        return NaN
```

Comment:
Since the error was not replicable, I'm wondering if we should add a console.log here and ask a user receiving the NaN-error to do it again and then check the logs. Might be an OS-related error for iOS maybe?

Reply:
We can add a log here and i have added the errorhandling to the caller. When the number is still NaN, the consumer is sent back to the shop page, because with the checks in place the NaN error should only happen, when the user did something we didn't planed to happen.

For example the go to viva wallet and use the back button in the browser triggers some errors in the backend etc.
It's time for the frontend tests ;)

```
    const sum = shopStore.calculateSum()
    if (sum > 0) {
      paymentStore.setPrice(sum)
    } else {
      // when the sum is still 0 or NaN, then something is wrong with the items in the shop
      router.push({ name: 'Shop' })
    }
```

Reply:
You are right. The correct way would be to add an integrated test here instead of a log to check how to reproduce the error. If the redirection to the shop page acts like an error catch i.e. log, I'm fine with this and we can merge it :)